gr-qtgui/include/gnuradio/qtgui/matrix_sink.h

```diff
@@ -0,0 +1,87 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2023 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#ifndef INCLUDED_QTGUI_MATRIX_SINK_H
+#define INCLUDED_QTGUI_MATRIX_SINK_H
+
+#include <gnuradio/qtgui/api.h>
+#include <gnuradio/sync_block.h>
+#include <qapplication.h>
+#include <QWidget>
+
+namespace gr {
+namespace qtgui {
+
+/*!
+ * \brief A graphical sink that displays a matrix.
+ * \ingroup qtgui
+ * \ingroup qtgui_blk
+ *
+ * \details
+ * This block displays a matrix as a 2D plot. The matrix is
+ * displayed as a contour plot, with the color of each contour
+ * determined by the value of the matrix at that point.
+ *
+ * The matrix is displayed as a 2D plot, with the x and y axes
+ * corresponding to the row and column indices of the matrix,
+ * respectively. The z axis corresponds to the value of the
+ * matrix at that point.
+ *
+ */
+class QTGUI_API matrix_sink : virtual public gr::sync_block
+{
+public:
+    // gr::qtgui::matrix_sink::sptr
+    typedef std::shared_ptr<matrix_sink> sptr;
+
+    /*!
+     * \brief Build a matrix sink block.
+     *
+     * \param name The name of the block.
+     * \param num_cols The number of columns in the matrix.
+     * \param vlen The vector length of the matrix.
+     * \param contour Whether or not to display the matrix as a contour plot.
+     * \param color_map The color map to use for the contour plot.
+     * \param interpolation The interpolation method to use for the contour plot.
+     * \param x_start The starting value of the x axis.
+     * \param x_end The ending value of the x axis.
+     * \param y_start The starting value of the y axis.
+     * \param y_end The ending value of the y axis.
+     * \param z_max The maximum value of the z axis.
+     * \param z_min The minimum value of the z axis.
+     * \param x_axis_label The label for the x axis.
+     * \param y_axis_label The label for the y axis.
+     * \param z_axis_label The label for the z axis.
+     * \param parent The parent QWidget.
+     */
+    static sptr make(const std::string& name,
```

Comment:
Other widgets also hava a lot of contstructor params, but this is way too many. Let's have defaults for everything that has a setter, then call the setter for each param or set of params.

Reply:
got it